### DISCUSSION TITLE
Redis on Symfony 5.3.7

### DISCUSSION THREAD

**Discussion by karaka200** [symfony/symfony#42843] [Q&A] [answered]
Do I need to use Redis in a Symfony 5.3.7? ( my database is MySQL )

Is it reasonable to use Radis for a website (like [lemonade](https://www.lemonade.com))? [ instead MySql ]

Before it I use symfony default cache `Symfony\Contracts\Cache\CacheInterface` . redis is better than it?

**Comment by Nek-** [comment, marked as answer]
> Do I need to use Redis in a Symfony 5.3.7? ( my database is MySQL )

You don't **need**. You _can_.

> Is it reasonable to use Radis for a website (like lemonade)? [ instead MySql ]

Do you mean _redis_ as main storage? Then no, it's not "reasonable". But if they do so, they must have a system to backup their redis episodically, there's a bunch of tools doing it. With imagination and work, you can do anything.

> Before it I use symfony default cache Symfony\Contracts\Cache\CacheInterface . redis is better than it?

Symfony Cache can work with many different provider, redis is one of them but you can even use your filesystem (it's just slower). Symfony does **not enforce** the use of redis. Use your favorite tool (Memcached is a great alternative for many cases). All cache databases will be "hidden" by Symfony and you will use the `CacheInterface` for any of them! This way you can switch your cache provider by just tweaking the Symfony configuration and not your code!
